### PR TITLE
Miasma improvements

### PR DESCRIPTION
**Pull request by Elijahrane**
## About the PR 
Cleanup, bugfixes, lots of audiovisual detail.
**Screenshots**

**Changelog**

:cl: Rane
- add: Corpses are now circled by buzzing flies when they've started to rot.
- tweak: Do you know how heavy cows are? It's wack. They're expensive to transport, and the cow crate price has been increased.
- fix: Freezer crates now stop rotting.
- fix: Rejuvenation properly removes rotting.
- fix: Body parts should gib from rotting for real this time.

### CODE REVIEW

File: Content.Server/Administration/Commands/RejuvenateCommand.cs [hunk 1]
```diff
@@ -98,6 +99,11 @@ public static void PerformRejuvenate(EntityUid target)
             {
                 EntitySystem.Get<DiseaseSystem>().CureAllDiseases(target, carrier);
             }
+
+            if (entMan.TryGetComponent<RottingComponent>(target, out var rotting))
+            {
+                entMan.RemoveComponent<RottingComponent>(target);
+            }
```
Comment (mirrorcult) [inline Content.Server/Administration/Commands/RejuvenateCommand.cs:106]:
```suggestion
            entMan.RemoveComponent<RottingComponent>(target);
```

remcomp checks for this already

File: Resources/Prototypes/Entities/Mobs/NPCs/simplemob.yml [hunk 2]
```diff
@@ -151,6 +151,12 @@
   - type: MobPrice
     price: 1000 # Living critters are valuable in space.
   - type: Perishable
+  - type: AmbientSound #Ideally in the future this will support multiple sources
+    enabled: false
+    volume: -5
+    range: 3
+    sound:
+      path: /Audio/Ambience/Temporary/flies.ogg
```
Comment (mirrorcult) [inline Resources/Prototypes/Entities/Mobs/NPCs/simplemob.yml:159]: yeah this is definitely sus right now. making the flies a separate entity is probably the best solution overall but need another opinion on that

**Review by mirrorcult — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Entities/Mobs/NPCs/simplemob.yml (hunk 2 above)
Comment (Elijahrane) [reply Resources/Prototypes/Entities/Mobs/NPCs/simplemob.yml:159]: Mmmm, would an entity that just holds the ambient sound stuff that gets parented to entities with FliesComponent and removed when it's shut down be decent?

File: Content.Server/Atmos/Miasma/MiasmaSystem.cs [hunk 3]
```diff
@@ -77,36 +98,77 @@ private void OnGibbed(EntityUid uid, PerishableComponent component, BeingGibbedE
...
+                ToggleDecomposition(args.Entity, true, perishable);
+        }
+
+
+        /// Fly stuff
+
+        private void OnFliesInit(EntityUid uid, FliesComponent component, ComponentInit args)
+        {
+            component.VirtFlies = EntityManager.SpawnEntity("AmbientSoundSourceFlies", Transform(uid).Coordinates);
+            Transform(component.VirtFlies).ParentUid = uid;
```
Comment (mirrorcult) [inline Content.Server/Atmos/Miasma/MiasmaSystem.cs:139]:
```suggestion
            Transform(component.VirtFlies).AttachParent(uid);
```

File: Resources/Audio/Ambience/Temporary/license.txt [hunk 4]
```diff
@@ -0,0 +1 @@
+flies.ogg taken from https://freesound.org/people/telezon/sounds/321870/
```
Comment (mirrorcult) [inline Resources/Audio/Ambience/Temporary/license.txt:1]:
```suggestion
flies.ogg taken from https://freesound.org/people/telezon/sounds/321870/ licensed under CC-0
```

File: Resources/Prototypes/Entities/Effects/ambient_sounds.yml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+- type: entity
+  id: AmbientSoundSourceFlies
+  name: ambient sound source
+  suffix: flies
```
Comment (mirrorcult) [inline Resources/Prototypes/Entities/Effects/ambient_sounds.yml:4]:
```suggestion
- type: entity
  id: AmbientSoundSourceFlies
  noSpawn: true
```

File: Resources/Prototypes/Entities/Effects/ambient_sounds.yml [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+  components:
+  - type: AmbientSound
+    volume: -5
+    range: 3
+    sound:
+      path: /Audio/Ambience/Temporary/flies.ogg
```
Comment (mirrorcult) [inline Resources/Prototypes/Entities/Effects/ambient_sounds.yml:10]:
```suggestion
  - type: AmbientSound
    volume: -5
    range: 3
    sound:
      path: /Audio/Ambience/Temporary/flies.ogg
  - type: Tag
    tags:
    - HideContextMenu
```

shouldn't happen normally but best to do this for all the abstract-type entities like this

File: Resources/Prototypes/Entities/Mobs/Species/human.yml [hunk 7]
```diff
@@ -325,6 +325,12 @@
     price: 1500 # Kidnapping a living person and selling them for cred is a good move.
     deathPenalty: 0.01 # However they really ought to be living and intact, otherwise they're worth 100x less.
   - type: Perishable
+  - type: AmbientSound
+    enabled: false
+    volume: -5
+    range: 3
+    sound:
+      path: /Audio/Ambience/Temporary/flies.ogg
```
Comment (mirrorcult) [inline Resources/Prototypes/Entities/Mobs/Species/human.yml:333]: remove?

**Review by mirrorcult — CHANGES_REQUESTED** [review]

**Comment by Elijahrane** [comment]
Addressed reviews + verified in game:
1. Sound plays when the entity starts to produce miasma and has fly visuals
2. Sound source is correctly parented to the entity
3. Sound source goes away after rejuvanate, entity deleted, or component removed

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**